### PR TITLE
Implement current_exe for AIX

### PR DESCRIPTION
**Pull request by bzEq**
Added `current_exe` implementation for AIX. This implementation doesn't rely on system specific syscalls, I guess some other targets can also use it.

### CODE REVIEW

File: library/std/src/sys/unix/os.rs [hunk 1]
```diff
@@ -471,6 +471,43 @@ pub fn current_exe() -> io::Result<PathBuf> {
...
+    ))?;
+    let path = PathBuf::from(exe_path);
+    if path.is_absolute() {
+        return path.canonicalize();
+    }
+    // Search PWD to infer current_exe.
+    if let Some(pstr) = path.to_str() && pstr.contains("/") {
+        return getcwd().map(|cwd| cwd.join(path))?.canonicalize();
+    }
+    // Search PATH to infer current_exe.
```
Comment (Mark-Simulacrum) [inline library/std/src/sys/unix/os.rs:496]: Are we searching PATH and PWD on other platforms today? I'm not sure we should do that here -- it increases the amount of work done and if it's not done on other platforms is somewhat inconsistent (and could have false positives much more easily).

Comment (Mark-Simulacrum) [reply library/std/src/sys/unix/os.rs:496]: Right, but you're using argv[0] just above which is what we do on other platforms (AFAICT, looking in this file, for example).

I am inclined to leave it there unless there's strong motivation for doing this cwd/path traversal. current_exe is documented as fallible for a reason.

Comment (Mark-Simulacrum) [reply library/std/src/sys/unix/os.rs:488]: Apologies if I wasn't clear. I think we should *remove* the PATH searching from this function, just returning the passed argument. IOW, I think if just adding aix to the fuchsia function works, then that seems like the right initial implementation here.

**Comment by bzEq** [comment]
Hi @Mark-Simulacrum , this PR is subsumed by https://github.com/rust-lang/rust/pull/109882, could you please help review that PR?

**Pull request closed without merging**